flink-python/pyflink/common/state_backend.py

```diff
@@ -0,0 +1,797 @@
+                 j_rocks_db_state_backend=None):
+        """
+        Creates a new :class:`RocksDBStateBackend` that stores its checkpoint data in the given
+        state backend or the location of given URI.
+
+        If using state backend, typically, one would supply a filesystem or database state backend
+        here where the snapshots from RocksDB would be stored.
+
+        If using URI, a state backend that stores checkpoints in HDFS or S3 must specify the file
+        system host and port in the URI, or have the Hadoop configuration that describes the file
+        system (host / high-availability group / possibly credentials) either referenced from the
+        Flink config, or included in the classpath.
+
+        :param checkpoint_data_uri: The URI describing the filesystem and path to the checkpoint
+                                    data directory.
+        :param enable_incremental_checkpointing: True if incremental checkpointing is enabled.
+        :param checkpoint_stream_backend: The backend write the checkpoint streams to.
+        :param j_rocks_db_state_backend: For internal use, please keep none.
+        """
+        if j_rocks_db_state_backend is None:
+            gateway = get_gateway()
+            JTernaryBoolean = gateway.jvm.org.apache.flink.util.TernaryBoolean
+            JRocksDBStateBackend = gateway.jvm.org.apache.flink.contrib.streaming.state \
+                .RocksDBStateBackend
+
+            if enable_incremental_checkpointing not in (None, True, False):
+                raise TypeError("Unsupported input for 'enable_incremental_checkpointing': %s, "
+                                "the value of the parameter should be None or"
+                                "True or False.")
+
+            if checkpoint_data_uri is not None:
+                if enable_incremental_checkpointing is None:
+                    j_rocks_db_state_backend = JRocksDBStateBackend(checkpoint_data_uri)
+                else:
+                    j_rocks_db_state_backend = \
+                        JRocksDBStateBackend(checkpoint_data_uri, enable_incremental_checkpointing)
+            elif isinstance(checkpoint_stream_backend, StateBackend):
+                if enable_incremental_checkpointing is None:
+                    j_enable_incremental_checkpointing = JTernaryBoolean.UNDEFINED
+                elif enable_incremental_checkpointing is True:
+                    j_enable_incremental_checkpointing = JTernaryBoolean.TRUE
+                else:
+                    j_enable_incremental_checkpointing = JTernaryBoolean.FALSE
+
+                j_rocks_db_state_backend = \
+                    JRocksDBStateBackend(checkpoint_stream_backend._j_state_backend,
+                                         j_enable_incremental_checkpointing)
+
+        self._j_rocks_db_state_backend = j_rocks_db_state_backend
+        super(RocksDBStateBackend, self).__init__(j_rocks_db_state_backend)
+
+    def configure(self, config):
```

Comment:
The method `configure` is used internally and can be removed. What do you think?

Reply:
That makes sense, I'll remove it in subsequent commit.